File: 455-assign-cookies/assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()

        contentChildCount = 0
        pointerG = 0
        pointerS = 0
        while pointerG < len(g) and pointerS < len(s):
            if s[pointerS] >= g[pointerG]:
                contentChildCount += 1
                pointerG += 1
            
            pointerS += 1
        return contentChildCount
        

File: 455-assign-cookies/assign-cookies.ts

    function findContentChildren(g: number[], s: number[]): number {
        g.sort((a, b) => a - b);
        s.sort((a, b) => a - b);

        let contentChildCount = 0;
        let pointerG = 0;
        let pointerS = 0;
        // Iterate using two pointers
        while (pointerG < g.length && pointerS < s.length) {
            // If the current cookie satisfies the current child's greed
            if (s[pointerS] >= g[pointerG]) {
                contentChildCount++;
                pointerG++;
            }
            pointerS++;
        }
        return contentChildCount;
    }

